docs/getting_started.rst

```diff
@@ -394,6 +407,10 @@ and ``couchdb_admin_party`` context helpers.
 
         print('Databases: {0}'.format(client.all_dbs()))
 
+.. warning:: Don't use the ``doc.delete()`` method inside your ``Document``
+             context manager. This method immediately deletes the document on
+             the server.
```

Comment:
The documentation for deleting a doc is pretty light http://python-cloudant.readthedocs.io/en/stable/getting_started.html#delete-a-document
I feel like if we're saying don't delete it this way then we ought to document an alternative deletion method. e.g. 
i) Does setting `_deleted: true` within the context work such that the save on leaving the context cause the update style delete? In which case that would be a workaround for the document context.
ii) Presumably calling `doc.delete` from a DB context works, so perhaps that is the context manager approach to deletion to document. Are there others at that level too  e.g. `db.pop('docid')`?

We should probably also extend this to say something like "and the document save when leaving the context will re-create an empty document with the document ID". Otherwise someone might be tempted to call `doc.delete()` in the document context anyway because they don't care if it happens immediately or not if you catch my meaning.